client/common/common.c

```diff
@@ -598,6 +606,28 @@ menu_with_options(struct client *client)
 enum bm_run_result
 run_menu(const struct client *client, struct bm_menu *menu, void (*item_cb)(const struct client *client, struct bm_item *item))
 {
+    if (!client->multi_instance_mode) {
+        char *xdg_runtime_dir = getenv("XDG_RUNTIME_DIR");
+        
+        char *bemenu_lock_location;
+        char buffer[128];
+        if (xdg_runtime_dir == NULL) {
+            bemenu_lock_location = "/tmp/bemenu.lock";
+            strcpy(buffer, bemenu_lock_location);
+        } else {
+            strcpy(buffer, xdg_runtime_dir);
+            strcat(buffer, "/bemenu.lock");
+        }
+        
+        int menu_lock = open(buffer, O_CREAT | O_RDWR | O_CLOEXEC, 0666); // Create a read-write lock file(if not already exists), closes on exec. Creates in tmp due to permissions.
+        errno = 0;
```

Comment:
This isn't needed as we don't check the errno here. Can get rid of errno.h include as well then